build/dsc/templates/dsc_type.rb.erb

```diff
@@ -5,6 +5,18 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
 
   provide :powershell, :parent => Puppet::Type.type(:base_dsc).provider(:powershell) do
     defaultfor :operatingsystem => :windows
+
+  <% if resource.ensurable? -%>
+    def ensure_value
```

Comment:
Some issue here causing the generation to be moved in too far. It comes out as:

~~~ruby
    def ensure_value
    'present'
  end
~~~